### DISCUSSION TITLE
输入域名www.test.com，默认到VUE的http://www.test.com/pigeon/index.html#/Fund/Index这个页面，怎么配置呀？

### DISCUSSION THREAD

**Discussion by willzhujian** [bfenetworks/bfe#819] [Q&A (问题和回答)] [answered]
mod_rewrite里面配置
{
                "Cond": "req_path_suffix_in(\"/\", false)",
                "Actions": [
                    {
                        "Cmd": "PATH_PREFIX_ADD",
                        "Params": [
                            "pigeon/index.html#/Fund/Index"
                        ]
                    }
                ],
                "Last": true
            },

后台请求日志
![image](https://user-images.githubusercontent.com/54895023/132489859-9f22da2c-1690-4450-b757-54323870db46.png)
浏览器显示页面不存在

**Comment by iyangsj** [comment]
404错误是后端实例返回的（无法正确处理bfe转发的请求）

检查下后端实例配置是否正确（例如，直接发送请求是否返回预期响应）
`
curl -H "Host: www.test.com" http://<BACKEND_ADDR>/pigeon/index.html#/Fund/Index
`

**Reply by willzhujian** [reply]
直接在浏览器中输入，肯定可以了；你看看日志他把“#”转换成了"%23"；正常浏览器输入日志是http://www.test.com/pigeon/index.html；后面#开始不记录的

**Comment by iyangsj** [comment, marked as answer]
浏览器发送请求时，请求uri中并不会包含 #/Fund/Index （#是用来指导浏览器动作的）
- 后端实例看起来可以正确处理 Host为www.test.com且URI为 pigeon/index.html的请求，而无法处理 Host为www.test.com且URI为 pigeon/index.html#/Fund/Index 或 pigeon/index.html%23/Fund/Index 的请求

建议：
1. 方式一、在mod_rewrite配置中去掉Params包含#的部分
2. 方式二、使用mod_redirect模块（参见 https://www.bfe-networks.net/zh_cn/modules/mod_redirect/mod_redirect/ ）
注：其中Cmd设置为URL_SET, Params 设置为 http://www.test.com/pigeon/index.html#/Fund/Index

**Reply by willzhujian** [reply]
恩，这样可以，就是浏览器链接重定向变了
